tests/testyamux.nim

```diff
@@ -179,6 +180,63 @@ suite "Yamux":
       writerBlocker.complete()
       await streamA.close()
 
+    asyncTest "Increase window size":
+      mSetup(512000)
+      let readerBlocker = newFuture[void]()
+      yamuxb.streamHandler = proc(conn: Connection) {.async.} =
+        await readerBlocker
+        var buffer: array[260000, byte]
+        discard await conn.readOnce(addr buffer[0], 260000)
+        await conn.close()
+
+      let streamA = await yamuxa.newStream()
+      check streamA == yamuxa.getStreams()[0]
+
+      await wait(streamA.write(newSeq[byte](512000)), 1.seconds) # shouldn't block
+
+      let secondWriter = streamA.write(newSeq[byte](10000))
+      await sleepAsync(10.milliseconds)
+      check: not secondWriter.finished()
+
+      readerBlocker.complete()
+      await wait(secondWriter, 1.seconds)
+
+      await streamA.close()
+
+    asyncTest "Reduce window size":
```

Comment:
I'm not sure this test is necessary. Shouldn't the purpose of this PR be to allow a max window size bigger than the default one?  I believe we don't want to make the max window size smaller than the default one.

Reply:
ditto